pages/apis/javascript/latest/belt.mdx

```diff
@@ -120,6 +120,55 @@ For example, Belt has the following set modules:
 
 ## Implementation Details
 
+### Array access type safety
+
+One common confusion comes from the way Belt handles array access. It differs from than the default standard library's.
+
+```reason
+let letters = [|"a", "b", "c"|];
+let a = letters[0];
+let capitalA = Js.String2.toUpperCase(a);
+let k = letters[10]; /* Raises an exception! The 10th index doesn't exist. */
+```
+
+Because Belt avoids exceptions and returns `options` instead, this code behaves differently:
+
+```reason
+open Belt;
+let letters = [|"a", "b", "c"|];
+let a = letters[0];
+let captialA = Js.String2.toUpperCase(a); /* Type error! This code will not compile. */
+let k = letters[10]; /* k == None */
+```
+
+Although we've fixed the problem where `k` raises an exception, we now have a type error when trying to capitalize `a`. There are a few things going on here:
+
+- Reason transforms array index access to the function `Array.get`. So `letters[0]` is the same as `Array.get(letters, 0)`. 
+- The compiler uses whichever `Array` module is in scope. If you `open Belt`, then it uses `Belt.Array`.
+- `Belt.Array.get` returns values wrapped in options, so `letters[0] == Some("a")`. 
+
+Fortunately, this is easy to fix:
+
+```reason
```

Comment:
is it possible to do `reason example` for verifying the example in our tests?

Reply:
Done!